### PR TITLE
Migrate view_image.jsx to be follow the guidline

### PR DESCRIPTION
**Pull request by CometKim**
#### Summary
Migrate view_image.jsx to be follow the guidline.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/7684

#### Checklist
- [x] Added or updated unit tests
- [x] Has UI changes

### CODE REVIEW

File: components/view_image.jsx [hunk 1]
```diff
@@ -318,18 +345,6 @@ export default class ViewImageModal extends React.Component {
...
-    startIndex: 0
-};
-ViewImageModal.propTypes = {
-    show: PropTypes.bool.isRequired,
-    onModalDismissed: PropTypes.func.isRequired,
-    fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-    startIndex: PropTypes.number
-};
-
 function LoadingImagePreview({progress, loading}) {
```
Comment (saturninoabril) [inline components/view_image.jsx]: I suggest the following to be on a separate file as a separate pure component with corresponding component tests:
- LoadingImagePreview to loading_image_preview.jsx
- ImagePreview to image_preview.jsx

File: tests/components/view_image.test.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            <ViewImageModal
+                show={true}
+                onModalDismissed={jest.fn()}
+                fileInfos={[fileInfo]}
+            />
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+});
```
Comment (saturninoabril) [inline tests/components/view_image.test.jsx]: Test for `ViewImageModal` is a bit low. You may find related samples at `tests/components`. I'd suggest to at least cover the following:
- handleNext
- handlePrev
- handleKeyPress
- onModalShown
- onModalHidden
- componentWillReceiveProps
- showImage
- loadImage
- handleImageLoaded
- handleImageProgress
- handleGetPublicLink
- onMouseEnterImage
- onMouseLeaveImage

screen shot 2017-10-26 at 11 05 40 am

screen shot 2017-10-26 at 11 11 54 am

**Comment by CometKim** [comment]
Thank you @saturninoabril @jwilander for your review!

I will take a look for it soon :smile:

File: components/view_image.jsx (hunk 1 above)
Comment (CometKim) [reply components/view_image.jsx]: @saturninoabril Separating component files is a good idea. They don't have a state, so can I leave it in a stateless-functional style, not a pure component?

File: components/view_image.jsx [hunk 3]
```diff
@@ -198,7 +211,6 @@ export default class ViewImageModal extends React.Component {
                 content = (
                     <ImagePreview
                         fileInfo={fileInfo}
-                        fileUrl={fileUrl}
```
Comment (CometKim) [inline components/view_image.jsx]: Hey @saturninoabril @jwilander. 

I just moved this line into child component which I separated. Should I do this same for all other preview components, or restore?

Comment (saturninoabril) [reply components/view_image.jsx]: No need. If necessary, separate ticket will be created for it.

File: components/view_image.jsx (hunk 1 above)
Comment (saturninoabril) [reply components/view_image.jsx]: Yes, it'd be fine as stateless functional component.

**Comment by saturninoabril** [comment]
@CometKim You have failing test, at least in `make check-style`:
screen shot 2017-10-30 at 10 53 26 pm

**Comment by CometKim** [comment]
I don't know why modal component of react-bootstrap makes errors on enzyme rendering...

**Comment by saturninoabril** [comment]
@CometKim I'll take a look at this today.

**Comment by saturninoabril** [comment]
@CometKim I have pushed one this commit, adding/fixing tests.

**Comment by CometKim** [comment]
Amazing!! Thank you @saturninoabril 
I'm not used to writing tests with enzyme yet. So much learned from your code. 😍 

(BTW, I want to see how tests work, but the github is too slow to my internet now)

**Comment by jwilander** [comment]
Nice work on this @CometKim! I'll merge after the 4.4 release branch gets cut

**Pull request merged**